python: format overflow in logging

The `python` plugin produces format overflow warnings with GCC due to nullptr in the arguments.

```
In file included from ../src/include/kdberrors.h:14,
                 from ../src/plugins/python/python.cpp:30:
../src/plugins/python/python.cpp: In function ‘int libelektra_python_LTX_elektraPluginOpen(ckdb::Plugin*, ckdb::Key*)’:
../src/include/kdblogger.h:113:37: warning: ‘%s’ directive argument is null [-Wformat-overflow=]
  113 | #define ELEKTRA_LOG(...) elektraLog (ELEKTRA_LOG_LEVEL_INFO, __func__, __FILE__, __LINE__, __VA_ARGS__)
      |                          ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/include/kdberrors.h:80:17: note: in expansion of macro ‘ELEKTRA_LOG’
   80 |                 ELEKTRA_LOG ("Add Error %s: " reason, ELEKTRA_ERROR_INSTALLATION, __VA_ARGS__);                                            \
      |                 ^~~~~~~~~~~
../src/plugins/python/python.cpp:294:25: note: in expansion of macro ‘ELEKTRA_SET_INSTALLATION_ERRORF’
  294 |                         ELEKTRA_SET_INSTALLATION_ERRORF (errorKey, "Unable to extend sys.path with script dirname '%s'", dname);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/include/kdblogger.h:113:37: warning: ‘%s’ directive argument is null [-Wformat-overflow=]
  113 | #define ELEKTRA_LOG(...) elektraLog (ELEKTRA_LOG_LEVEL_INFO, __func__, __FILE__, __LINE__, __VA_ARGS__)
      |                          ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/include/kdberrors.h:80:17: note: in expansion of macro ‘ELEKTRA_LOG’
   80 |                 ELEKTRA_LOG ("Add Error %s: " reason, ELEKTRA_ERROR_INSTALLATION, __VA_ARGS__);                                            \
      |                 ^~~~~~~~~~~
../src/plugins/python/python.cpp:268:25: note: in expansion of macro ‘ELEKTRA_SET_INSTALLATION_ERRORF’
  268 |                         ELEKTRA_SET_INSTALLATION_ERRORF (errorKey, "Unable to extend sys.path with user-defined /python/path '%s'", mname);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[299/375] Building CXX object src/plugins/python/CMakeFiles/elektra-python.dir/python.cpp.o
In file included from ../src/include/kdberrors.h:14,
                 from ../src/plugins/python/python.cpp:30:
../src/plugins/python/python.cpp: In function ‘int libelektra_python_LTX_elektraPluginOpen(ckdb::Plugin*, ckdb::Key*)’:
../src/include/kdblogger.h:113:37: warning: ‘%s’ directive argument is null [-Wformat-overflow=]
  113 | #define ELEKTRA_LOG(...) elektraLog (ELEKTRA_LOG_LEVEL_INFO, __func__, __FILE__, __LINE__, __VA_ARGS__)
      |                          ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/include/kdberrors.h:80:17: note: in expansion of macro ‘ELEKTRA_LOG’
   80 |                 ELEKTRA_LOG ("Add Error %s: " reason, ELEKTRA_ERROR_INSTALLATION, __VA_ARGS__);                                            \
      |                 ^~~~~~~~~~~
../src/plugins/python/python.cpp:294:25: note: in expansion of macro ‘ELEKTRA_SET_INSTALLATION_ERRORF’
  294 |                         ELEKTRA_SET_INSTALLATION_ERRORF (errorKey, "Unable to extend sys.path with script dirname '%s'", dname);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/include/kdblogger.h:113:37: warning: ‘%s’ directive argument is null [-Wformat-overflow=]
  113 | #define ELEKTRA_LOG(...) elektraLog (ELEKTRA_LOG_LEVEL_INFO, __func__, __FILE__, __LINE__, __VA_ARGS__)
      |                          ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/include/kdberrors.h:80:17: note: in expansion of macro ‘ELEKTRA_LOG’
   80 |                 ELEKTRA_LOG ("Add Error %s: " reason, ELEKTRA_ERROR_INSTALLATION, __VA_ARGS__);                                            \
      |                 ^~~~~~~~~~~
../src/plugins/python/python.cpp:268:25: note: in expansion of macro ‘ELEKTRA_SET_INSTALLATION_ERRORF’
  268 |                         ELEKTRA_SET_INSTALLATION_ERRORF (errorKey, "Unable to extend sys.path with user-defined /python/path '%s'", mname);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -67,6 +67,12 @@ The following section lists news about the [plugins](https://www.libelektra.org/
 - <<TODO>>
 - <<TODO>>
 
+### Python
+
+- Fix format string overflow and add error checking when appending to `sys.path`. _(Mihael Pranjić)_
+- <<TODO>>
+- <<TODO>>
+
 ## Libraries
 
 The text below summarizes updates to the [C (and C++)-based libraries](https://www.libelektra.org/libraries/readme) of Elektra.

--- a/src/plugins/python/python.cpp
+++ b/src/plugins/python/python.cpp
@@ -81,7 +81,10 @@ static int Python_AppendToSysPath (const char * path)
 
 	PyObject * sysPath = PySys_GetObject ((char *) "path");
 	PyObject * pyPath = PyUnicode_FromString (path);
-	PyList_Append (sysPath, pyPath);
+	if (PyList_Append (sysPath, pyPath) == -1)
+	{
+		return 0;
+	}
 	Py_DECREF (pyPath);
 	return 1;
 }
@@ -265,6 +268,10 @@ int PYTHON_PLUGIN_FUNCTION (Open) (ckdb::Plugin * handle, ckdb::Key * errorKey)
 		const char * mname = keyString (ksLookupByName (elektraPluginGetConfig (handle), "/python/path", 0));
 		if (!Python_AppendToSysPath (mname))
 		{
+			if (!mname)
+			{
+				mname = "<nullptr>";
+			}
 			ELEKTRA_SET_INSTALLATION_ERRORF (errorKey, "Unable to extend sys.path with user-defined /python/path '%s'", mname);
 			goto error;
 		}
@@ -291,6 +298,10 @@ int PYTHON_PLUGIN_FUNCTION (Open) (ckdb::Plugin * handle, ckdb::Key * errorKey)
 		const char * dname = dirname (tmpScript);
 		if (!Python_AppendToSysPath (dname))
 		{
+			if (!dname)
+			{
+				dname = "<nullptr>";
+			}
 			ELEKTRA_SET_INSTALLATION_ERRORF (errorKey, "Unable to extend sys.path with script dirname '%s'", dname);
 			elektraFree (tmpScript);
 			goto error;

--- a/src/plugins/python/python.cpp
+++ b/src/plugins/python/python.cpp
@@ -81,10 +81,13 @@ static int Python_AppendToSysPath (const char * path)
 
 	PyObject * sysPath = PySys_GetObject ((char *) "path");
 	PyObject * pyPath = PyUnicode_FromString (path);
+
 	if (PyList_Append (sysPath, pyPath) == -1)
 	{
+		Py_DECREF (pyPath);
 		return 0;
 	}
+
 	Py_DECREF (pyPath);
 	return 1;
 }

